### DISCUSSION TITLE
Fish Configuration Improvements

### DISCUSSION THREAD

**Discussion by joshmedeski** [pnpm/pnpm#4830] [Ideas]
So I'm using pnpm with fish and the auto-complete script came out to this in my `config.fish` file:

```fish
# tabtab source for packages
# uninstall by removing these lines
[ -f ~/.config/tabtab/fish/__tabtab.fish ]; and . ~/.config/tabtab/fish/__tabtab.fish; or true
```

There are a couple of improvements I'd like to suggest and discuss.

1. Move the config file to the `~/.config/fish/completions` directory (it won't have to be called in `config.fish`)
2. Only use `pnpm.fish` and drop the `__tabtab.fish` file (simplify how many files are required to setup autocomplete)

Let me know your thoughts and I can make a pull request off of this discussion.

**Comment by zkochan** [comment]
We use a fork tabtab for autocompletion. If you can make this change, I don't have objections. But it would be good to keep that package a general-purpose package. So that others may use it as well.
